cmake/therock_artifacts.cmake

```diff
@@ -15,6 +15,10 @@ function(therock_provide_artifact slice_name)
     "COMPONENTS;SUBPROJECT_DEPS"
   )
 
+  if(${slice_name} MATCHES "_")
+    message(FATAL_ERROR "Artifact slice name '${slice_name}' is not allowed to contain an underscore")
+  endif()
```

Comment:
We could also use an allow-list instead of a deny-list, such as only allowing names matching `^[a-zA-Z][a-zA-Z0-9-]*$`

* Must start with a lowercase or uppercase letter
* May contain any number of additional lowercase or uppercase letters, numbers, and `-`
* May not contain any other characters